anyway/infographics_utils.py

```diff
@@ -1554,8 +1586,20 @@ def get_accidents_stats(
     # get stats
     query = get_query(table_obj, filters, start_time, end_time)
     if group_by:
-        query = query.group_by(group_by)
-        query = query.with_entities(group_by, func.count(count))
+        if isinstance(group_by, tuple):
+            if len(group_by) == 2:
+                query = query.group_by(*group_by)
+                query = query.with_entities(*group_by, func.count(count))
+                dd = query.all()
+                res = retro_dictify(dd)
+                return res
+            else:
+                err_msg = f'get_accidents_stats: {group_by}: Only a string or a tuple of two are valid for group_by'
```

Comment:
Why do we allow grouping by 1-2 fields only? Is it because of the generation of the two level dict?
I don't think we have use cases of grouping by 3 right now - but we might in the future (btw did you make sure we don't have such cases?)

However - I think we should take care of such cases when they arrive and keep your code as is. 

_**I think it's more important to merge this pr than to think of hypothetic use cases that are not relevant right now**_

Reply:
OK.
The code that generates the nested dict should work for multiple levels, though I am not sure I understand it. I found it on the network. The need to fill empty items is more complicated so I only implemented it for two levels.